scipy/special/_basic.py

```diff
@@ -3415,22 +3358,4 @@ def zeta(x, q=None, out=None):
     if q is None:
         return _ufuncs._riemann_zeta(x, out)
     else:
-        return _ufuncs._zeta(x, q, out)
-
-
-def _sph_harm_all(m, n, theta, phi):
-    """Private function. This may be removed or modified at any time."""
-
-    theta = np.asarray(theta)
-    if (not np.issubdtype(theta.dtype, np.inexact)):
-        theta = theta.astype(np.float64)
-
-    phi = np.asarray(phi)
-    if (not np.issubdtype(phi.dtype, np.inexact)):
-        phi = phi.astype(np.float64)
-
-    out = np.empty((2 * m + 1, n + 1) + np.broadcast_shapes(theta.shape, phi.shape),
-        dtype = np.result_type(1j, theta.dtype, phi.dtype))
-    _sph_harm_all_gufunc(theta, phi, out = np.moveaxis(out, (0, 1), (-2, -1)))
-
-    return out
+        return _ufuncs._zeta(x, q, out)
```

Comment:
I think we should configure the lint CI job to check for blank lines at the end of files.

Reply:
done in gh-21023